File: aurastream-backend/src/main/java/com/aurastream/backend/dto/PlaylistDto.java
package com.aurastream.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PlaylistDto {
    private Long id;
    private String name;
    private String description;
    private String coverUrl;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<PlaylistItemDto> items;
    private int totalSongs;
}

File: aurastream-mobile/app/src/main/java/com/aurastream/mobile/data/dto/PlaylistDto.kt
package com.aurastream.mobile.data.dto

import com.google.gson.annotations.SerializedName
import com.aurastream.mobile.domain.model.Playlist
import com.aurastream.mobile.domain.model.PlaylistItem

data class PlaylistItemDto(
    @SerializedName("id") val id: Long,
    @SerializedName("song") val song: SongDto,
    @SerializedName("addedAt") val addedAt: String?,
    @SerializedName("position") val position: Int
)

data class PlaylistDto(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String?,
    @SerializedName("coverUrl") val coverUrl: String?,
    @SerializedName("createdAt") val createdAt: String?,
    @SerializedName("items") val items: List<PlaylistItemDto>?,
    @SerializedName("totalSongs") val totalSongs: Int
)

fun PlaylistItemDto.toDomain(baseUrl: String): PlaylistItem = PlaylistItem(
    id = id,
    song = song.toDomain(baseUrl),
    addedAt = addedAt ?: "",
    position = position
)

fun PlaylistDto.toDomain(baseUrl: String): Playlist = Playlist(
    id = id,
    name = name,
    description = description,
    coverUrl = coverUrl ?: "https://images.unsplash.com/photo-1518609878373-06d740f60d8b",
    createdAt = createdAt,
    items = items?.map { it.toDomain(baseUrl) } ?: emptyList(),
    totalSongs = totalSongs
)
